geometric_kernels/spaces/special_orthogonal_group.py

```diff
@@ -0,0 +1,408 @@
+"""
+Special orthogonal group, SO(n).
```

Comment:
Code in this file is very hard to properly review. Is there a paper we can link and match some of the equations in there to bits of the code?

Reply:
I agree, the code is hard to read. Unfortunately there's no publicly accessible paper to check against